aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1360,19 +1369,244 @@ Tensor nuclear_norm(const Tensor& self, IntArrayRef dim, bool keepdim) {
 
 Tensor& nuclear_norm_out(Tensor& result, const Tensor& self, IntArrayRef dim, bool keepdim) {
   TORCH_CHECK(dim.size() == 2, "nuclear norm requires a 'dim' argument of size 2");
+  auto dim_ = dim.vec();
+  maybe_wrap_dims(dim_, self.dim());
 
-  auto permutation = create_dim_backshift_permutation(dim[0], dim[1], self.dim());
+  auto permutation = create_dim_backshift_permutation(dim_[0], dim_[1], self.dim());
   auto permutation_reverse = create_reverse_permutation(permutation);
 
   Tensor p = self.permute(permutation);
   at::sum_out(result, std::get<1>(at::svd(p, /*some=*/true, /*compute_uv=*/false)), -1, keepdim);
   if (keepdim) {
     result.unsqueeze_(-1);
-    result = result.permute(permutation_reverse);
+    Tensor result_ = result.permute(permutation_reverse);
+    at::native::resize_output(result, result_.sizes());
```

Comment:
hm, does this constantly warn now? `result` is non-empty (as it is result of sum operation), and presumably with the wrong sizes, or you would not need the resize.

Reply:
@kurtamohler, you are right, previous handling of `out` was wrong, `result.permute()` changes `result` reference, and unfortunately we don't have `permute_`. But you can use set_ or as_strided_ on original result_ to avoid resizing and copy that you do here.

Reply:
Can you please add a test for this behavior that you fix, to make sure it does not regress?

Reply:
Done, I've changed it to `set_` and added a unit test.